Add model data handling/loading

busykoala opened the pull request:
The model is responsible to handle the data in a convenient way. It has to provide a nice interface so that all the logic is abstracted when using.

The functionality implemented in this PR are:
- conditionally loading the most recent data (if there is new data)
- updating the data stored in the annotations if necessary

Commit:
Check if there is a more recent report.

If there is data in the annotations a check is done if a more recent report exists in the filelisting block configured. If that's the case this data is loaded into a df.

mbaechtold commented on ftw/linkchecker/browser/dashboard.py:
```diff
@@ -20,6 +20,18 @@ def __init__(self, context, request):
+        self._reports = self._load_and_sort_fileblock_reports()
+
+        self._latest_report_df = self._get_latest_report_df_from_excel()
+        if isinstance(self._latest_report_df, pd.DataFrame):
+            # there is new data -> update/merge data
+            pass
+        else:
+            # no data or no new data found
+            pass
```
This block does nothing, so I would remove it.

busykoala replied:
The PR was not yet ready to review. I'm still implementing that ;-)

mbaechtold replied:
Oooops, my bad 🙈

Commit:
Add data updating functionality.

Commit:
Add new data and handle case if no data is available.

Commit:
Remove test covered by others implicitly.

This test failed because the test data has changed. It is covered by the functional tests of the API though and removing it does not add more risk.
